### PR TITLE
Expose an CLI option of post script in advanced reboot test to user

### PR DESCRIPTION
**Pull request by stephenxs**
Signed-off-by: Stephen Sun <stephens@nvidia.com>

### Description of PR

Summary:
Expose an CLI option of post script in advanced reboot test to user

### Type of change

- [ ] Bug fix
- [ ] Testbed and Framework(new/improvement)
- [x] Test case(new/improvement)

#### What is the motivation for this PR?
Expose a CLI option by which the user can designate the script to run after the warm reboot to check something.
For example, in the dynamic buffer calculation, the buffer mode (traditional mode/dynamic model) needs to be checked after warm reboot, which can be done by this script.

### CODE REVIEW

File: tests/common/fixtures/advanced_reboot.py [hunk 1]
```diff
@@ -501,6 +502,10 @@ def tearDown(self):
 
         self.__runScript(['remove_ip.sh'], self.ptfhost)
 
+        if self.postRebootCheckScript:
+            logger.info('Run the post reboot check script')
+            self.__runScript([self.postRebootCheckScript], self.duthost)
```
Comment (neethajohn) [inline tests/common/fixtures/advanced_reboot.py:507]: Where is the code that copies this script to the dut?

Comment (stephenxs) [reply tests/common/fixtures/advanced_reboot.py:507]: It’s done here [advanced_reboot.py#L219](https://github.com/Azure/sonic-mgmt/blob/581944481768f8457366aaecfc99983022af3696/tests/common/fixtures/advanced_reboot.py#L219)

File: tests/platform_tests/args/advanced_reboot_args.py [hunk 2]
```diff
@@ -84,3 +84,11 @@ def add_advanced_reboot_args(parser):
         default=False,
         help="Replace fast-reboot script on DUT",
     )
+
+    parser.addoption(
+        "--post_reboot_check_script",
+        action="store",
+        type=str,
+        default=None,
+        help="Script for checking additional states on DUT"
```
Comment (neethajohn) [inline tests/platform_tests/args/advanced_reboot_args.py:93]: Will you be checking in the script via a separate PR?

Comment (stephenxs) [reply tests/platform_tests/args/advanced_reboot_args.py:93]: Yes. The ability to designate a user specified post script is provided here.
After that, additional scripts can be provided in other PRs

**Review by neethajohn — APPROVED** [review]
